### ISSUE TITLE
result of builder does not conform to MangedService

### ISSUE THREAD

**Issue by nadavwe** [wix/wix-embedded-mysql#44]
Hey @viliusl,
while working with the Idans (@idankWix @idancohen11), we wanted to use the embedded mysql with the TestEnv support.
unfortunately, the builder has a ```start``` method, and the server has a ```stop``` method.
i think the builder should have a build that creates an object with start & stop and then it can be used directly as a ```MangedService``` with the implicits @hugebdu created.
https://github.com/wix-platform/wix-framework/blob/0f46e4ab99bdc485b10317f3a59387ae22599e81/hoopoe-common/hoopoe-it-test/src/main/scala/com/wixpress/framework/test/env/ManagedService.scala

WDYT?

**Comment by noam-almog** [comment]
@nadavwe due to the builder pattern implemented here i think it would be hard to do.
for your convinience you can also use this: https://github.com/wix/mysql-testkit
it has a factory method that will comply with the managed service api.
note that this is an open sourced project.

**Comment by ittaiz** [comment]
@nadavwe these are exactly the kinds of things that should be in the "wix"
part IMO

On Thu, Mar 24, 2016 at 2:45 PM Nadav Wexler <notifications@github.com>
wrote:

> Hey @viliusl <https://github.com/viliusl>,
> while working with the Idans (@idankWix <https://github.com/idankWix>
> @idancohen11 <https://github.com/idancohen11>), we wanted to use the
> embedded mysql with the TestEnv support.
> unfortunately, the builder has a start method, and the server has a stop
> method.
> i think the builder should have a build that creates an object with start
> & stop and then it can be used directly as a MangedService with the
> implicits @hugebdu <https://github.com/hugebdu> created.
>
> https://github.com/wix-platform/wix-framework/blob/0f46e4ab99bdc485b10317f3a59387ae22599e81/hoopoe-common/hoopoe-it-test/src/main/scala/com/wixpress/framework/test/env/ManagedService.scala
>
> WDYT?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/wix/wix-embedded-mysql/issues/44>
>
-- 

*Ittai Zeidman*

Head of Server Guild/Core Server



Cell: 054-6735021

40 Hanamal street, Tel Aviv, Israel


<http://www.wix.com/>

**Comment by viliusl** [comment]
@nadavwe yes, EmbeddedMysql does not conform to a startable/stoppable service notion and there are several reasons for that:
 - idea of this testkit is to be used in an e2e set-up where database should be started once per test-suite, so contract is made to signal this;
 - implementation is not actually restartable due to all extractions and other thing. I could do it restartable in the end of the day, but not worth the effort;
- less state handling given there is no start()/stop() on the service - what happens if you start started or stop stopped or stop not started etc.

So yes, I suppose you cannot use implicits, but why not create explicits?:)

**Comment by nadavwe** [comment]
* it should start only once. this is why i would put it in the ITEnvironment. i don't need it restartable.
* if you use the InitGuardedManagedService, you get this stop/start conditions handled for you. 
* the reason to use implicits is just so this won't be (compile-time) coupled to the ManagedService code. but maybe it's a better idea to actually couple it.

**Comment by viliusl** [comment]
@nadavwe yes, but what you are talking about is wix-bound library/code. wix-embedded-mysql is an OSS/unbounded lib that was intended to be used without wixy infra.

Now what we are thinking of is having an adapter (ManagedService) that glues this library to be pluggable into ITEnvironment and other things we do use. And In this case we will need to couple to ManagedService as current impl will not work with existing implicits:/

**Comment by viliusl** [comment]
Closing, as this issue is out of scope for wix-embedded-mysql
